### ISSUE TITLE
Make `Rav1dPicAllocator` fully safe

### ISSUE THREAD

**Issue by kkysen** [memorysafety/rav1d#1009]
This should be possible, leaving the only `unsafe` in the `Dav1dPicAllocator` conversions, but it's fairly tricky, so we can save it for later.

**Comment by rinon** [comment]
Is this done now?

**Comment by kkysen** [comment]
> Is this done now?

No.  I wasn't planning on doing this unless someone wanted it.  The current API is `unsafe`, while what this issue suggested was a rewrite to make the API fully safe from Rust.

**Comment by rinon** [comment]
Ah I see, I think that could go along with a safe Rust API, if you're only using the C API then would it even make sense to try to make this fully safe?

**Comment by kkysen** [comment]
> Ah I see, I think that could go along with a safe Rust API, if you're only using the C API then would it even make sense to try to make this fully safe?

Yeah that would make sense to go together.  You're right, the C API wouldn't get safer from this (though internally we'd use the Rust API for the default, and that's what usually used).
